src/compute/arithmetics/basic/div.rs

```diff
@@ -109,10 +111,28 @@ where
 /// ```
 pub fn div_scalar<T>(lhs: &PrimitiveArray<T>, rhs: &T) -> PrimitiveArray<T>
 where
-    T: NativeType + Div<Output = T>,
+    T: NativeType + Div<Output = T> + NumCast,
 {
     let rhs = *rhs;
-    unary(lhs, |a| a / rhs, lhs.data_type().clone())
+    match T::DATA_TYPE {
+        DataType::UInt64 => {
+            //  Safety:
+            // We need to go from generic to concrete type and back to generic.
+            // We know the types are the same, but the compiler needs some convincing
+            unsafe {
```

Comment:
remove 1 `unsafe`:

```
           let lhs = (lhs as &dyn Array)
                .as_any()
                .downcast_ref::<PrimitiveArray<u64>>()
                .unwrap();
            let rhs = rhs.to_u64().unwrap();
            let reduced_div = StrengthReducedU64::new(rhs);
            let result = unary(lhs, |a| a / reduced_div, lhs.data_type().clone());
            // Safety: we just proved that `lhs` is `PrimitiveArray<u64>` which means that
            // T = u64
            unsafe { std::mem::transmute::<PrimitiveArray<u64>, PrimitiveArray<T>>(result) }
```

Reply:
:+1:

Reply:
Does this mean the "transmute-free" claim must go?

Reply:
Transmute-lean should be the goal😁

Reply:
at least when I wrote "transmute-free", I was thinking about `arrow-rs` treating every buffer as `&[u8]` and converting them to `&[T]` without being able to prove (only trust) that the buffers are indeed created from `&[T]` (and valid for some size N).

We do use transmutes in the crate atm; e.g. the IPC has [one transmute](https://github.com/jorgecarleitao/arrow2/blob/main/src/io/ipc/read/read_basic.rs#L69) when reading bytes to `&mut [T]`, because there is no way of doing this in safe Rust (despite being `safe`).

imo this is another of those cases: the way to achieve this in safe rust is via trait specialization, which is only available in the unstable channel.

The relevant aspect imo is that we are in control of the call and know exactly why it is safe.

Reply:
But maybe there is a way to achieve the same without transmutes; I would definitely prefer that!